docs/docs/building-a-site-with-authentication.md

```diff
@@ -2,33 +2,84 @@
+Or from a third party like:
 
-- [Create client-only routes](/tutorial/authentication-tutorial/#creating-client-only-routes),
-  to tell Gatsby which routes should be rendered on demand
-- [Wrap private content in a PrivateRoute component](/tutorial/authentication-tutorial/#controlling-private-routes),
-  to check if a user is authenticated or not, therefore rendering the content or
-  redirecting to another page (usually, the login page)
+- Firebase
+- Auth0
+- AWS Amplify
+- Netlify Identity
+
+The process of verifying a user on the client against an authentication service and using a token returned by it to get protected data is similar for any of these services and is visualized in this diagram:
+
+![Diagram of Gatsby using an authentication service to get data from an API](./images/basic-auth.png)
+
+1. First, a request is made from a Gatsby site (the client) to an authentication service to perform an action like register a new user or login.
+
+2. If the credentials (like a username and password) provided from the client match a user in the authentication service, it returns a token (like a JSON Web Token, abbreviated as JWT) so that the user has a key that they can use to prove they are who they say they are. The data returned about the user can be stored in the Gatsby app and passed to components using a Provider component with the React Context API using the [`wrapRootElement` API](/docs/browser-apis/#wrapRootElement) from Gatsby.
+
+3. With the key, the client can make a request to an external data source like an API (the server) where protected data is stored, the key is unique to a specific user that allows them to access their specific data.
+
+4. The server returns data back to the client that it can use on the client to pass information into components.
+
+_**Note**: this is the same pattern that other sites built with React (like Create React App) would need to follow._
+
+## Implementing authentication in a Gatsby site
+
+There are a few things to be aware of when implementing authentication in a Gatsby site, because of how Gatsby uniquely builds pages and renders static assets with dynamic capabilities.
+
+### Setting up client-only routes
+
+With Gatsby, you are able to create restricted areas in your app using [client-only routes](/docs/building-apps-with-gatsby/#client-only-routes).
+
+Because static HTML files generated by Gatsby sit on a file server, you cannot programmatically control access to those files on the server from Gatsby (for example: a user could guess or type in a URL and navigate straight to the page). As the [section from the Adding App and Website functionality](/docs/adding-app-and-website-functionality/#client-only-routes) overview page demonstrates, client-only routes can be created to route a user between pages using a React-based router, as opposed to navigating between different static HTML files on a server. Gatsby is different from a traditional Single Page Application in
```

Comment:
```suggestion
Because static HTML files generated by Gatsby sit on a file server, you cannot programmatically control access to those files (for example: a user could guess or type in a URL and navigate straight to the page). As the [section from the Adding App and Website functionality](/docs/adding-app-and-website-functionality/#client-only-routes) overview page demonstrates, client-only routes can be created to route a user between pages using a React-based router, as opposed to navigating between different static HTML files on a server. Gatsby is different from a traditional Single Page Application in
```

Reply:
Think something is missing at the end of this paragraph?

Reply:
Updated this and added a little section back in the overview about differences between Gatsby and "traditional" React apps.